### ISSUE TITLE
Add ability to pass AbstratSchema instances to Encoder

### ISSUE THREAD

**Issue by PabloKowalczyk** [neomerx/json-api#168]
Hi,

currently we can pass FQCN or Closure (that return AbstractSchema instance) to Encoder - it is good but is this possible to add option to support already created Schema instances?

It will be very useful in Symfony projects where Schemas are services.

Example:
```php
class Foo
{
  private $encoder;

  public function __construct(CommentSchema $commentSchema)
  {
    $this->encoder = Encoder::instance([
      Comment::class => $commentSchema
    ]);
  }
}
```

**Comment by neomerx** [comment]
Closure that returns the instance? I understand it's not an equivalent of just passing instances. My first reaction 'why not?' though If I didn't do it in the first place it might be the reason. I need to check the code if it has issues.

**Comment by PabloKowalczyk** [comment]
Yes, i mean "Closure that returns the instance". I edited original issue.

**Comment by neomerx** [comment]
OK.

Just if anybody search for similar issue. Possible solution
```php
    public function __construct(CommentSchema $commentSchema)
    {
        $this->encoder = Encoder::instance([
            Comment::class => $this->createGetter($commentSchema),
        ]);
    }
    public function createGetter($value)
    {
       return function () use ($value) {
           return $value;
       };
    }
```

**Comment by PabloKowalczyk** [comment]
This is possible solution but can we pass Schema instance without wrapping it in Closure?

**Comment by neomerx** [comment]
Check it out in `develop` branch and let me know if it's what you have expected. Then I'll merge to `master` and release.

**Comment by neomerx** [comment]
seems to work though due to https://github.com/travis-ci/travis-ci/issues/7226 travis fails on PHP 5.6

**Comment by neomerx** [comment]
a workaround for travis issue found https://github.com/travis-ci/travis-ci/issues/7226#issuecomment-275855616

**Comment by PabloKowalczyk** [comment]
It's working as expected. Thank you.

**Comment by PabloKowalczyk** [comment]
My bad. We can pass already created Schema instance to Encoder but Encoder throw `InvalidArgumentException: Schema is not registered for resource type` when i try to call `encode()` method on Encoder.

**Comment by neomerx** [comment]
I added a test that reproduces the error and made necessary amendments in code. Check the update and let me know.

**Comment by PabloKowalczyk** [comment]
Now everything work. Thank you.

**Comment by neomerx** [comment]
Doc updated. Merged to master.

**Comment by neomerx** [comment]
Released as v0.8.11
